app/code/community/IyzicoCheckoutForm/IyzicoCheckoutForm/Model/Method/Abstract.php

```diff
@@ -0,0 +1,96 @@
+<?php
+abstract class IyzicoCheckoutForm_IyzicoCheckoutForm_Model_Method_Abstract extends Mage_Payment_Model_Method_Abstract {
+
+    protected $_isGateway = true;
+    protected $_canUseCheckout = true;
+    protected $_canUseForMultishipping = false;
+    protected $_isInitializeNeeded = false;
+    protected $_methodCode = '';
+    protected $_methodTitle = '';
+    protected $_code = 'iyzicocheckout_abstract';
+    protected $_implementation = 'iframe';
+    protected $_canCapture = true;
+    protected $_canRefund = true;
+    protected $_canRefundInvoicePartial = true;
+
+    public function getCredentials() {
+
+        $credentials = array(
+            'api_id' => Mage::getStoreConfig('payment/' . $this->getCode() . '/api_id', $this->getOrder()->getStoreId()),
+            'secret' => Mage::getStoreConfig('payment/' . $this->getCode() . '/secret_key', $this->getOrder()->getStoreId()),
+            'prefix' => Mage::getStoreConfig('payment/' . $this->getCode() . '/prefix', $this->getOrder()->getStoreId()),
+        );
+        return $credentials;
+    }
+
+    public function getOrder() {
+        $paymentInfo = $this->getInfoInstance();
+
+        if ($paymentInfo instanceof Mage_Sales_Model_Order_Payment) {
+            return $paymentInfo->getOrder();
+        }
+
+        return $paymentInfo->getQuote();
+    }
+
+    public function getOrderPlaceRedirectUrl() {
+        return Mage::getSingleton('customer/session')->getRedirectUrl();
+    }
+
+    public function authorize() {
+        try {
+            $currentCurrencyCode = Mage::app()->getStore()->getCurrentCurrencyCode();
+            $currencyArr = Mage::helper('iyzicocheckoutform')->getCurrencyArr();
+            if (!in_array($currentCurrencyCode, $currencyArr)) {
+                Mage::throwException(Mage::helper('iyzicocheckoutform')->__("Please make sure that the currency value in your payment request is one of the (USD, EUR, GBP, IRR, TL) valid values"));
+            }
+            $response = Mage::getSingleton('iyzicocheckoutform/iyzicoCheckoutForm_request', array($this->getOrder()))->request();
+            if ('make_iyzico_api_call_false' != $response) {
+                if ("success" == $response->getStatus()) {
+                    if ('iframe' == $this->_getImplementation()) {
+                        $this->_iframe($response);
+                    } else {
+                        $this->_redirect($response);
+                    }
+                } else {
+                    $errorMsg = $response->getErrorMessage();
+                    if (empty($errorMsg)) {
+                        $errorMsg = 'There are some internal error, please try again';
```

Comment:
Burdaki hata olmayacak hatalara verişmiş ve geçiştirilmiş yani seninde anladığın gibi iyzico dan eğer işlem başarısız ise error mesajı dönmeme ihtimali yok ama yinede bir kontrol yaptırılmış mesela burda ödeme alındımı  durumunun successmi diye sorgulamış böyle bir hata olmasına imkan yok istersen bunların hepsi için bana bir türkçe bir mesaj gönder hepsine aynısını yazıyım ben bunları gördüm fakat güncellemedim önemsiz diye